go/tools/wtool/main.go

```diff
@@ -0,0 +1,144 @@
+/*wtool augments your bazel WORKSPACE file with new_go_repository entries
+
+Example Usage: wtool com_github_golang_glog com_google_cloud_go
+will add 2 new_go_repository to your WORKSPACE
+by converting com_github_golang_glog -> github.com/golang/glog
+and so forth and then doing a 'git ls-remote' to get
+the latest commit.
+
+if wtool cannot figure out the bazel -> Go mapping, try
+Other Usage: wtool -asis github.com/golang/glog
+which takes an importpath, and computes the bazel name + ls-remote as above.
+*/
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	bzl "github.com/bazelbuild/buildifier/core"
+	"github.com/bazelbuild/rules_go/go/tools/gazelle/rules"
+	"github.com/bazelbuild/rules_go/go/tools/gazelle/wspace"
+	"golang.org/x/tools/go/vcs"
+)
+
+var (
+	asis = flag.Bool("asis", false, "if true, leave the import names as-is (by default they are treated as bazel converted names like org_golang_x_net")
+)
+
+func main() {
+	flag.Parse()
+	if err := run(flag.Args()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(args []string) error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	w, err := wspace.Find(cwd)
+	if err != nil {
+		return err
+	}
+	p := filepath.Join(w, "WORKSPACE")
+	b, err := ioutil.ReadFile(p)
+	if err != nil {
+		return err
+	}
+	f, err := bzl.Parse(p, b)
+	if err != nil {
+		return err
+	}
+	for _, arg := range args {
+		imp, err := findImport(arg)
+		if err != nil {
+			return err
+		}
+		f.Stmt = append(f.Stmt, imp)
+	}
+	bzl.Rewrite(f, nil)
+	return ioutil.WriteFile(f.Path, bzl.Format(f), 0644)
+}
+
+func nameAndImportpath(name string) (string, string, error) {
+	if *asis {
+		return rules.ImportPathToBazelRepoName(name), name, nil
+	}
+	s := strings.Split(name, "_")
+	if len(s) < 4 {
+		return "", "", fmt.Errorf("only 4-part strings supported: %q", name)
+	}
+	rest := strings.Join(s[3:], "-")
+	if strings.HasPrefix(name, "org_golang_google") {
+		return name, "google.golang.org/" + rest, nil
+	}
+	if strings.HasPrefix(name, "com_google_cloud") {
+		return name, "cloud.google.com/" + rest, nil
+	}
+	return name, strings.Join([]string{s[1] + "." + s[0], s[2], rest}, "/"), nil
```

Comment:
I special cased google.golang.org and cloud.google.com, we could make that configurable, or again, allow the user to use -asis